README.md

```diff
@@ -297,6 +298,35 @@ The `inputData` field is a JSON object that stores data from the user's input as
 progresses. This field is placed in the model handed to the Thymeleaf templates, so each page should
 have access to it.
 
+## Actions
+
+Actions (like [conditions](#conditions)) are defined in Java as methods, and can read from the `currentSubmission` object.
+
+```java
+public class CalculateBeforeSave implements Action {
+  
+  float RATE = 0.59;
+  
+  public void run(Submission submission) {
+    int mileage = submission.getInputData()
+        .get('mileage');
+    submission.getInputData()
+        .put('reimbursement', mileage * RATE);
+  }
+} 
+```
+
+### beforeSave
+
+An action can be added to a page in the flow-config to update the submission data before saving to the database.
+
+```yaml
+expense:
+  beforeSave: actions.CalculateBeforeSave
```

Comment:
I like how this looks, but really it needs to be a full path to the class: https://github.com/codeforamerica/form-flow-starter-app/blob/main/src/main/resources/flows-config.yaml#L49

Maybe this could be `java.path.to.actions.CalculateBeforeSave`.  This could link to the starter app example in the link here. 

I'd love to figure out how to make this just the class name, but couldn't figure out how to reference the `actionsPath` created earlier in the yaml file: https://github.com/codeforamerica/form-flow-starter-app/blob/main/src/main/resources/flows-config.yaml#L3

Reply:
Agreed! Not sure how to do that rn tho - will keep it in mind